File: java/socketd/src/main/java/org/noear/socketd/transport/core/entity/MessageBuilder.java
package org.noear.socketd.transport.core.entity;

import org.noear.socketd.transport.core.Constants;
import org.noear.socketd.transport.core.Entity;
import org.noear.socketd.transport.core.Flags;
import org.noear.socketd.transport.core.MessageInternal;

/**
 * 消息构建器
 *
 * @author noear
 * @since 2.1
 */
public class MessageBuilder {
    private int flag = Flags.Unknown;
    private String sid = Constants.DEF_SID;
    private String event = Constants.DEF_EVENT;
    private Entity entity = null;

    /**
     * 设置标记
     */
    public MessageBuilder flag(int flag) {
        this.flag = flag;
        return this;
    }

    /**
     * 设置流id
     */
    public MessageBuilder sid(String sid) {
        this.sid = sid;
        return this;
    }

    /**
     * 设置事件
     */
    public MessageBuilder event(String event) {
        this.event = event;
        return this;
    }

    /**
     * 设置实体
     */
    public MessageBuilder entity(Entity entity) {
        this.entity = entity;

        return this;
    }

    /**
     * 构建
     */
    public MessageInternal build() {
        return new MessageDefault(flag, sid, event, entity);
    }
}

File: python/socketd/socketd/transport/core/entity/MessageBuilder.py
from __future__ import annotations

from socketd.transport.core import Entity
from socketd.transport.core.Costants import Constants
from socketd.transport.core.Flags import Flags
from socketd.transport.core.entity.MessageDefault import MessageDefault


class MessageBuilder:
    def __init__(self):
        self._flag = Flags.Unknown
        self._sid = Constants.DEF_SID
        self._event = Constants.DEF_EVENT
        self._entity = None

    def flag(self, flag: int):
        self._flag = flag
        return self

    def sid(self, sid: str):
        self._sid = sid
        return self

    def event(self, event: str):
        self._event = event
        return self

    def entity(self, entity: Entity):
        self._entity = entity
        return self

    def build(self) -> MessageDefault:
        return MessageDefault(self._flag, self._sid, self._event, self._entity)
